### DISCUSSION TITLE
EarlyBlendOut is not work well when lag on network.

### DISCUSSION THREAD

**Discussion by watsonsong** [Sixze/ALS-Refactored#581] [Q&A]
Open the network simulate to BAD and will see the position correction when montage has RootMotion and with AlsAnimNotifyState_EarlyBlendOut.

If the client predicate play montage and the the server start to play montage, the client will met the EarlyBlendOut before the server, so client will move forward and corrected back by server, because the server is still in root motion so it can not move by input.

If the server play montage first(Eg, GAS with server initialize), the server will met the EarlyBlendOut before the client. The client send the move input to server(That why EarlyBlendOut could work). The server will move first but the client is still in root motion. So the client will corrected by server.

Maybe the server should not move directly by move input, if the client can not response the move input. For example the client in a root motion. This will avoid the second case. And for the first case, the EarlyBlendOut may need start a little early than client.

**Comment by Sixze** [comment]
Unfortunately I can't do anything about it now, as things like this should be handled at the GAS level, but I'm not ready to integrate it yet, so I can only advise you to avoid using this animated notification.
